port python 2 to 3

hstarmans opened the pull request:
Tried your code on python 3 but couldn't read from sdram.
Should be fixed now, tested it on ubuntu 16.04 with xula2lx25

rheineke committed:
Readme update

rheineke committed:
Tk skeleton

rheineke committed:
Port test frame stubbed

rheineke committed:
PNG added to Tk GUI

rheineke committed:
Tk improvements

rheineke committed:
Merge branch 'python3' into tk

hstarmans committed:
changed / to // and forced uint to be int

changes needed to read from sdram;
	import xstools.xsboard
	fpga=xstools.xsboard.Xula2lx25(0)
	result=fpga.read_sdram(0,13)

rheineke commented:
Thanks for the bug fix. Can you write a simple unit/integration test to show the expected result(s)? Something like test/test_sdram.py:

```
import unittest

import xstools.xsboard


class TestSDRAM(unittest.TestCase):
    def test_read_sdram(self):
        fpga = xstools.xsboard.Xula2lx25(0)
        result = fpga.read_sdram(0, 13)
        expected = 'foo'
        self.assertEqual(result, expected)
```

hstarmans committed:
write to sdram

modified write to sdram and added a unit test for it

hstarmans committed:
changed round back to \\

hstarmans commented:
I have added the test as requested. The code should work with the xula2lx25. I do, however, get a deprecation warning:
DeprecationWarning: tostring() is deprecated. Use tobytes() instead. hexfile.gets(bottom,num_bytes).encode()).
This warning should be related to line 99 of ramdev.py, where I added encode(); not sure why this happens.

rheineke commented:
@hstarmans Thanks for the test. I merged your changes in with just a few formatting improvements.

I see the DeprecationWarning as well and it is coming from IntelHex. The maintainers fixed this but the fix doesn't appear to be in the latest release (2.1): https://github.com/bialix/intelhex/commit/4722987857589263a7a1358522065307d4b92fa7

Do you want to follow up with them about a new release?
